themes/classic/templates/catalog/product-miniature.tpl

```diff
@@ -12,7 +12,7 @@
 
     <div class="product-description">
       {block name='product_name'}
-        <h1 class="h3 product-title" itemprop="name"><a href="{$product.url}">{$product.name}</a></h1>
+        <h1 class="h3 product-title" itemprop="name"><a href="{$product.url}">{$product.name|truncate:30:'...'}</a></h1>
```

Comment:
Would `text-overflow: ellipsis` work here? (I know it only works in a very limited subset of cases and I'm usually disappointed)